docs/root/guides/package-names.md

```diff
@@ -0,0 +1,50 @@
+## Package names
+
+The package names for the Google Cloud Client Libraries for .NET take
+multiple forms. Here are a few examples:
+
+- Google.Cloud.Datastore.V1
+- Google.Cloud.Firestore
+- Google.Cloud.Logging.Type
+- Google.Cloud.Logging.Log4Net
+
+It's somewhat unusual to see a version number in a NuGet package
+name, as in the Datastore example. There's a reason though: this is
+a package for that specific version of the Datastore API. We
+distinguish between the *API* (exposed by Google servers, used by
+multiple languages) and the *package* (the .NET-specific NuGet
+package).
+
+The version number in the package name is the API version number.
+Separately, that package can go through multiple versions. A change
+to the API will always mean a new package version, but the reverse
+isn't true: sometimes we may improve the package even without the
+API changing. (For example, adding more convenience methods.)
+
+Having the version number in the package name gives the advantage
+that if you want to migrate from one major API version to another,
+you don't have to do that for your whole codebase in a single step.
+You can take a dependency on Google.Cloud.AwesomeApi.V1 and
+Google.Cloud.AwesomeApi.V2 in the same project, migrating your code
+piecemeal.
```

Comment:
Might also be worth mentioning how this brings sanity to having dependencies that themselves depend on Google APIs.
I.e. Your project can have deps on two different packages that themselves depend on different API versions.
I.e. Your project doesn't have to use the same major API version as your deps.

Reply:
Good call, will do.